tests/k8st/tests/test_bgp_advert.py

```diff
@@ -165,7 +165,42 @@ def test_mainline(self):
         retry_until_success(lambda: self.assertIn(local_svc_ip, self.get_routes()))
         retry_until_success(lambda: self.assertNotIn(cluster_svc_ip, self.get_routes()))
 
-        # Delete nginx service
+        # Create a network policy that only accepts traffic from the external node
+        run("""kubectl exec -i -n kube-system calicoctl -- /calicoctl apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-tcp-80-ex
+  namespace: bgp-test
+spec:
+  selector: all()
+  types:
+  - Ingress
+  ingress:
+  - action: Allow
+    protocol: TCP
+    source: { nets: ["10.192.0.5/32"] }
+    destination: { ports: ["80"] }
+EOF
+""")
+
+        # Connectivity to nginx-local should always succeed
+        self.do_external_curl("kube-node-extra", local_svc_ip)
+
+        # Connectivity to nginx-cluster will rarely succeed, try to curl 10 times
+        try:
+          for i in range(10):
+            self.do_external_curl("kube-node-extra", cluster_svc_ip)
+          self.fail("external node should not be able to access cluster svc")
+        except subprocess.CalledProcessError:
+          pass
+
+        # Scale the local_svc to 2 replicas
+        self.scale_deployment(local_svc, self.ns, 2)
+        matchStr = local_svc_ip + " proto bird \n\tnexthop via 10.192.0.3  dev eth0 weight 1\n\tnexthop via 10.192.0.4  dev eth0 weight 1"
```

Comment:
It would be really nice to wrap this up into a helper function - `assert_ecmp_routes` or something like that.

e.g. `self.assert_ecmp_routes(dst, via=[])`

Then, we could use it more generally to assert routes are balanced across multiple vias.